recipe/build.sh

```diff
@@ -7,21 +7,34 @@ if [[ ${cuda_compiler_version} != "None" ]]; then
   # Set the CUDA arch list from
   # https://github.com/conda-forge/pytorch-cpu-feedstock/blob/2be0b38024b3b5601fcefce40596fc2a5fce4ab7/recipe/build_pytorch.sh#L94
 
-  if [[ ${cuda_compiler_version} == 10.* ]]; then
-    export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5+PTX"
+  if [[ ${cuda_compiler_version} == 9.0* ]]; then
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;7.0+PTX"
+      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
+  elif [[ ${cuda_compiler_version} == 9.2* ]]; then
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0+PTX"
+      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
+  elif [[ ${cuda_compiler_version} == 10.* ]]; then
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5+PTX"
+      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 11.0* ]]; then
-    export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0+PTX"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0+PTX"
+      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 11.1 ]]; then
-    export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 11.2 ]]; then
-    export TORCH_CUDA_ARCH_LIST="6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 11.8 ]]; then
-    export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9+PTX"
+      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9+PTX"
+      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 12.0 ]]; then
-    export TORCH_CUDA_ARCH_LIST="5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
+      export TORCH_CUDA_ARCH_LIST="5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
+      # $CUDA_HOME not set in CUDA 12.0. Using $PREFIX
+      export CUDA_TOOLKIT_ROOT_DIR="${PREFIX}"
   else
-    echo "Unsupported cuda version. edit build.sh"
-    exit 1
+      echo "unsupported cuda version. edit build_pytorch.sh"
```

Comment:
Would suggest keeping `build.sh` (given that is where this list is kept in this feedstock)

```suggestion
      echo "unsupported cuda version. edit build.sh"
```